builtInFunctions/crossChainTokenChecker.go

```diff
@@ -8,13 +8,19 @@ import (
 )
 
 type crossChainTokenChecker struct {
-	selfESDTPrefix []byte
+	selfESDTPrefix       []byte
+	whiteListedAddresses map[string]struct{}
 }
 
 // NewCrossChainTokenChecker creates a new cross chain token checker
-func NewCrossChainTokenChecker(selfESDTPrefix []byte) (*crossChainTokenChecker, error) {
+func NewCrossChainTokenChecker(selfESDTPrefix []byte, whiteListedAddresses map[string]struct{}) (*crossChainTokenChecker, error) {
 	ctc := &crossChainTokenChecker{
-		selfESDTPrefix: selfESDTPrefix,
+		selfESDTPrefix:       selfESDTPrefix,
+		whiteListedAddresses: whiteListedAddresses,
```

Comment:
make sure that whitelistedAddresses is initialised - do not have as nil. line 61 will actually result in panic.

Reply:
This check is going to be done in the node integration: https://github.com/multiversx/mx-chain-go/pull/6177